src/main/scala/com/lightbend/kafkalagexporter/PrometheusEndpointSink.scala

```diff
@@ -15,45 +16,48 @@ object PrometheusEndpointSink {
   type ClusterName = String
   type GlobalLabels = Map[String, String]
   type ClusterGlobalLabels = Map[ClusterName, GlobalLabels]
+  type Metrics = Map[GaugeDefinition, Gauge]
 
   def apply(definitions: MetricDefinitions, metricWhitelist: List[String], clusterGlobalLabels: ClusterGlobalLabels,
             server: HTTPServer, registry: CollectorRegistry): MetricsSink = {
-    Try(new PrometheusEndpointSink(definitions, metricWhitelist, server, registry))
+    Try(new PrometheusEndpointSink(definitions, metricWhitelist, clusterGlobalLabels, server, registry))
       .fold(t => throw new Exception("Could not create Prometheus Endpoint", t), sink => sink)
   }
 }
 
-class PrometheusEndpointSink private(definitions: MetricDefinitions, metricWhitelist: List[String],
+class PrometheusEndpointSink private(definitions: MetricDefinitions, metricWhitelist: List[String], clusterGlobalLabels: ClusterGlobalLabels,
                                      server: HTTPServer, registry: CollectorRegistry) extends MetricsSink {
-
   DefaultExports.initialize()
 
-  private val metrics: Map[GaugeDefinition, Gauge] = register()
-
-  private def register(): Map[GaugeDefinition, Gauge] = {
-    definitions
-      .filter(d => metricWhitelist.exists(d.name.matches))
-      .map { d =>
-        d -> Gauge
-          .build()
-          .name(d.name)
-          .help(d.help)
-          .labelNames(d.labels: _*)
-          .register(registry)
-      }
-      .toMap
+  private val metrics: Metrics = {
+    val globalLabels = clusterGlobalLabels.values.headOption.getOrElse(Map.empty).keys.toSeq
+    definitions.filter(d => metricWhitelist.exists(d.name.matches)).map { d =>
+      d -> Gauge.build()
+        .name(d.name)
+        .help(d.help)
+        .labelNames(globalLabels ++ d.labels: _*)
+        .register(registry)
+    }.toMap
   }
 
+
   override def report(m: MetricValue): Unit = {
-    if(metricWhitelist.exists(m.definition.name.matches)) {
+    if (metricWhitelist.exists(m.definition.name.matches)) {
       val metric = metrics.getOrElse(m.definition, throw new IllegalArgumentException(s"No metric with definition ${m.definition.name} registered"))
-      metric.labels(m.labels: _*).set(m.value)
+      val globalLabelValuesForCluster = clusterGlobalLabels.getOrElse(m.clusterName, Map.empty)
+      metric.labels(globalLabelValuesForCluster.values.toSeq ++ m.labels: _*).set(m.value)
     }
   }
 
   override def remove(m: RemoveMetric): Unit = {
-    if(metricWhitelist.exists(m.definition.name.matches)) {
-      metrics.get(m.definition).foreach(_.remove(m.labels: _*))
+    if (metricWhitelist.exists(m.definition.name.matches)) {
+      for (
+        globalLabels <- clusterGlobalLabels.get(m.clusterName);
+        gauge <- metrics.get(m.definition)
+      ) {
```

Comment:
If you put your iterators in a `for { }` expression you don't need `;` to delimit them, you can use new lines instead.  

Ex)
```
for {
  globalLabels <- clusterGlobalLabels.get(m.clusterName)
  gauge <- metrics.get(m.definition)
}
```